commands.py

```diff
@@ -51,9 +64,23 @@ def wrap(self, *arg, **kwargs) -> Any:
     return decorator
 
 
-class CopilotTextCommand(LspTextCommand, metaclass=ABCMeta):
+class CopilotCommandBase(metaclass=ABCMeta):
     session_name = PACKAGE_NAME
+    requirement = REQUIRE_SIGN_IN
```

Comment:
I've been thinking about this API here.

Bitflags are useful when we need to support multiple conditions, but in our case, most likely we would never use anything else except `SIGN_IN`/`NOT_SIGN_IN`.  
So maybe we should rather go with a simple class variable:
```python
requires_sign_in: Optional[bool] = None
```
and when it `self.requires_sign_in is None` we don't do anything(even calling `CopilotPlugin.get_has_signed_in`), otherwise, we could do
```python
return True if self.requires_sign_in is None else self.requires_sign_in == CopilotPlugin.get_has_signed_in()
```
It is not critical, just thought it would make this API more obiouvs

Reply:
How can `requires_sign_in: Optional[bool]` represents `REQUIRE_NOT_SIGN_IN`?

Reply:
That makes the behavior more vague imho... How do you represents `REQUIRE_NOT_SIGN_IN` without introducing another variable?

Reply:
My idea was to use `False` to ensure sign out, `True` to ensure sign in, `None` to bypass the check altogether

Reply:
I feel it's a little vague that `requires_sign_in = False` means `require not sign in` or `not require sign in`.

Reply:
Agree 👍🏻